packages/iTwinUI-react/src/core/utils/components/Resizer.tsx

```diff
@@ -0,0 +1,298 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { getBoundedValue, getTranslateValues, getWindow } from '../functions';
+
+export type ResizerProps = {
+  /**
+   * Ref of the element that is being resized.
+   */
+  elementRef: React.RefObject<HTMLElement>;
+  /**
+   * Ref of the container element in order to avoid resizing past container boundaries.
+   * If not passed, viewport will be used.
+   */
+  containerRef?: React.RefObject<HTMLElement>;
+  /**
+   * Callback that is being called on resize end.
+   * Useful to preserve state if element is being closed.
+   */
+  setStyle?: (style: React.CSSProperties) => void;
+};
+
+/**
+ * Component that allows to resize parent element.
+ * Parent must have `position: relative`.
+ * @example
+ * const ref = React.useRef<HTMLDivElement>(null);
+ * return (
+ *   <div ref={ref} style={{ position: 'relative' }}>
+ *     <Resizer elementRef={ref} />
+ *   </div>
+ * );
+ */
+export const Resizer = (props: ResizerProps) => {
+  const { elementRef, containerRef, setStyle } = props;
+
+  const onResizePointerDown = (event: React.PointerEvent<HTMLElement>) => {
+    if (!elementRef.current || event.button !== 0) {
+      return;
+    }
+
+    const initialPointerX = event.clientX;
+    const initialPointerY = event.clientY;
+
+    const [initialTranslateX, initialTranslateY] = getTranslateValues(
+      elementRef.current,
+    );
+    const {
+      width: initialWidth,
+      height: initialHeight,
+    } = elementRef.current.getBoundingClientRect();
+
+    let width = `${initialWidth}px`;
+    let height = `${initialHeight}px`;
+    let translateX = initialTranslateX;
+    let translateY = initialTranslateY;
+
+    const minWidth = Number(
+      getComputedStyle(elementRef.current).minWidth.replace('px', ''),
+    );
+    const minHeight = Number(
+      getComputedStyle(elementRef.current).minHeight.replace('px', ''),
+    );
+
+    const resizer = event.currentTarget.dataset.iuiResizer;
+
+    const originalUserSelect =
+      elementRef.current.ownerDocument.body.style.userSelect;
+    elementRef.current.ownerDocument.body.style.userSelect = 'none';
+
+    const onResizePointerMove = (event: PointerEvent) => {
+      if (!elementRef.current) {
+        return;
+      }
+
+      const containerRect = containerRef?.current?.getBoundingClientRect();
+      const clientX = getBoundedValue(
+        event.clientX,
+        containerRect?.left ?? 0,
+        containerRect?.right ?? getWindow()?.innerWidth ?? 0,
```

Comment:
I think `document.documentElement.clientWidth` would be better than `window.innerWidth` here because `window.innerWidth` considers the "visual viewport" which can be affected things like zoom

Reply:
Changed